source/server/server.cc

```diff
@@ -130,6 +130,10 @@ InstanceImpl::~InstanceImpl() {
   listener_manager_.reset();
   ENVOY_LOG(debug, "destroyed listener manager");
 
+  // Stop the RTDS subscriptions before we shut everything down, as clearing the deferred list will
+  // otherwise invalidate the streams used by RTDS.
+  Runtime::LoaderSingleton::get().stop();
```

Comment:
nit: I think you have direct access to runtime and should use that vs. the singleton? Also, how does clearing the deferred delete list invalidate the streams if nothing has been stopped? I'm a little confused.

Reply:
I think it's because in `terminate()` above we shut down the cluster manager, which resets the `ads_mux_`, which would result in the grpc stream getting added to the list of deferred deletions. This whole shutdown ordering is definitely complex, so I can try adding some more comments.